src/rules/no-empty-fields.ts

```diff
@@ -0,0 +1,122 @@
+import type { AST as JsonAST } from "jsonc-eslint-parser";
+
+import * as ESTree from "estree";
+
+import { createRule, PackageJsonRuleContext } from "../createRule";
+
+const getDataAndMessageId = (
+	node:
+		| JsonAST.JSONArrayExpression
+		| JsonAST.JSONObjectExpression
+		| JsonAST.JSONProperty,
+): {
+	data: Record<string, string>;
+	messageId: "emptyExpression" | "emptyFields";
+} => {
+	switch (node.type) {
+		case "JSONArrayExpression":
+			return {
+				data: {
+					expressionType: "array",
+				},
+				messageId: "emptyExpression",
+			};
+		case "JSONObjectExpression":
+			return {
+				data: {
+					expressionType: "object",
+				},
+				messageId: "emptyExpression",
+			};
+		case "JSONProperty":
+			return {
+				data: {
+					field: (node.key as JsonAST.JSONStringLiteral).value,
+				},
+				messageId: "emptyFields",
+			};
+	}
+};
+
+const report = (
+	context: PackageJsonRuleContext,
+	node:
+		| JsonAST.JSONArrayExpression
+		| JsonAST.JSONObjectExpression
+		| JsonAST.JSONProperty,
+) => {
+	const { data, messageId } = getDataAndMessageId(node);
+	context.report({
+		data,
+		messageId,
+		node: node as unknown as ESTree.Node,
+		suggest: [
+			{
+				*fix(fixer) {
+					yield fixer.remove(node as unknown as ESTree.Node);
+
+					const tokenFromCurrentLine =
+						context.sourceCode.getTokenAfter(
+							node as unknown as ESTree.Node,
+						);
+					const tokenFromPreviousLine =
+						context.sourceCode.getTokenBefore(
+							node as unknown as ESTree.Node,
+						);
+
+					if (
+						tokenFromPreviousLine?.value === "," &&
+						tokenFromCurrentLine?.value !== ","
+					) {
+						yield fixer.remove(tokenFromPreviousLine);
+					}
+
+					if (tokenFromCurrentLine?.value === ",") {
+						yield fixer.remove(tokenFromCurrentLine);
+					}
```

Comment:
[Non-Actionable] This logic is another example of something that'll be simplified by #750. I'll go ahead and merge this PR now and factor the changes into #750.